docs/src/adr/04adr-reactor-stub.md

```diff
@@ -0,0 +1,78 @@
+# ADR-04: Atomkraft reactor stub command
+
+## Context
+This ADR describes one component of Atomkraft, as decribed in [ADR-01](01adr-principles-architecture.md).
+
+## Status
+This is an early version of the ADR, meant for the first prototype.
+Major changes to the scope are very likely.
+
+## Reactor Stub
+
+Once the user has a trace (either written by hand, or obtained from the system model),  they need to write a *reactor*.
+A reactor is a set of Python functions connecting the actions of the trace to executions of the code.
+
+In this ADR we describe a command `atomkraft reactor ...` which creates a stub of the reactor,
+auto-generating all the boilerplate, and leaving it to the user to only fill in the application-specific parts.
+
+## Command Line Interface
+
+The interface of the command looks like this:
+`atomkraft reactor <actions_list> <variables_list> [<reactor-stub-file>]`
+where
+ - `actions_list` is a list of actions for which to generate stubs
+ - `variables-list` is a list of all relevant state variables (those that could appear in traces of interest)
+ - `reactor_stub_file` is a path at which the reactor file should be created.
+ If omitted, a default path is used: `atomkraft/reactors/reactor.py`
+
+## Implementation
+
+ The `ReactorRoom` class implements the desired behavior.
+ Its member functions are:
+  - `generate_reactor(actions_list, variables_list, stub_file_path=None)`:
+  generates the stub, containing one `@step` function for each action from the `action_list`,
+  and each of these functions takes variables from the `variables_list` as arguments.
+  The reactor stub is created at `stub_file_path` name location
+  (if it is `None`, a default location is used).
+  - `check_reactor(trace, keypath, reactor=None)`: it checks if the `reactor` (default reactor if `None`) is compatible with the `trace`. A reactor is compatible with a trace if each action appearing in `trace` is matched with a function in `reactor`. The `keypath` argument defines which field of the ITF trace contains the action.
```

Comment:
the problem with having the function `check_reactor()` is that as its user (the `Execute` component), if I don't have a reactor (the end user didn't provide me such argument), I don't have anything to check. I can only kindly _ask_ to give me a reactor for the given trace; thus my suggestion to use `get_trace_reactor(trace, reactor = None)`

Reply:
I had a similar comment on your PR, but really suggesting the opposite.
My reasoning was that I will give a default reactor if it is not provided (that's why it has a default argument `None`).
If I do the same with `get_trace_reactor`, then providing the `reactor` doesn't make sense (because I would then always give the same path back). This is all conditioned on the assumption that we are dealing with paths here (and not some other kinds of objects).

Let me modify this passage to be more precise about the paths assumption and about what is the return value (that is completely missing now)

Reply:
Ok, I see:) so our functions, according to your description do basically the same thing. The problem with the naming of your function is that it violates one of the basic assumptions: `check_` functions always return a boolean. I suggest we rename it to `get_`; let's discuss online if you disagree